install_config/persistent_storage/pod_security_context.adoc

```diff
@@ -12,262 +12,728 @@ toc::[]
+prevent a container from being able to run as root.
+
+Because pods often match the *restricted* SCC, it is worth knowing the security
+this entails. The following observations apply to the *restricted* SCC:
+
+- Constrains user IDs due to the `*runAsUser*` strategy being set to
+*MustRunAsRange*. This forces user ID validation.
+- Because a range of allowable user IDs is not defined in the SCC (see `oc export
+scc restricted` for more details), the project's
+`*openshift.io/sa.scc.uid-range*` range will be used for range checking and for
+a default ID, if needed.
+- Causes a default user ID to be produced when a user ID is not specified in the
+pod definition due to `*runAsUser*` being set to *MustRunAsRange*.
+- requires a selinux label (`seLinuxContext` set to _MustRunAs_) which uses the namespace's
+default MCS label.
+- Allows arbitrary supplemental group IDs because no range checking is required.
+This is a result of both the `*supplementalGroups*` and `*fsGroup*` strategies
+being set to *RunAsAny*.
+- Produces no default supplemental groups for the running pod due to *RunAsAny*
+for the two group strategies above. Therefore, if no groups are defined in the
+pod definition (or in the image), the container(s) will have no supplemental
+groups predefined.
+
+The following shows the *default* project and a custom SCC (*my-custom-scc*),
+which summarizes the interactions of the SCC and the project:
+
+====
+----
+$ oc get project default -o yaml <1>
+...
+metadata:
+  annotations: <2>
+    openshift.io/sa.scc.mcs: s0:c1,c0 <3>
+    openshift.io/sa.scc.supplemental-groups: 1000000000/10000 <4>
+    openshift.io/sa.scc.uid-range: 1000000000/10000 <5>
+
+$ oc get scc my-custom-scc -o yaml <6>
+...
+fsGroup:
+  type: MustRunAs <7>
+  ranges:
+  - min: 5000
+    max: 6000
+runAsUser:
+  type: MustRunAsRange <8>
+  uidRangeMin: 99
+  uidRangeMax: 199
+seLinuxContext: <9>
+  type: MustRunAs
+  SELinuxOptions: <10>
+    user: <selinux-user-name>
+    role: ...
+    type: ...
+    level: ...
+supplementalGroups:
+  type: MustRunAs <7>
+  ranges:
+  - min: 5000
+    max: 6000
+----
+<1> *default* is the name of the project.
+<2> Default values are only produced when the corresponding SCC strategy is not
+*RunAsAny*.
+<3> SELinux default when not defined in the pod definition or in the SCC.
+<4> Range of allowable group IDs. ID validation only occurs when the SCC
+strategy is *RunAsAny*. There can be more than one range specified, separated by
+commas. Two range formats are supported: 1) `M/N`, where `M` is the starting ID
+and `N` is the count, so the range becomes `M` through (and including) `M+N-1`.
+2) `M-N`, where `M` is again the starting ID and `N` is the ending ID. The
+default group ID is the starting ID in the first range, which is `1000000000` in
+this project. If the SCC did not define a minimum group ID, then the project's
+default ID is applied.
+<5> Same as *<4>* but for user IDs. Also, only a single range of user IDs is
```

Comment:
These callouts are humongous.  Perhaps the common syntax (M/N, M-N) explanation could be put into  a small NOTE or somesuch.